monai/handlers/checkpoint_saver.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from ignite.engine import Events
+from ignite.handlers import ModelCheckpoint
+
+
+class CheckpointSaver:
+    """
+    CheckpointSaver acts as an Ignite handler to save checkpoint data into files.
+    It supports to save according to metrics result, epoch number, iteration number
+    and last model or exception.
+
+    Args:
+        save_dir (str): the target directory to save the checkpoints.
+        save_dict (dict): source objects that save to the checkpoint. examples::
+
+            {'network': net, 'optimizer': optimizer, 'engine', engine}
+
+        name (str): identifier of logging.logger to use, if None, defaulting to ``engine.logger``.
+        file_prefix (str): prefix for the filenames to which objects will be saved.
+        save_final (bool): whether to save checkpoint or session at final iteration or exception.
+        save_key_metric (bool): whether to save checkpoint or session according to key metric value.
+        key_metric_name (str): the name of key_metric in ignite metrics dictionary.
```

Comment:
docstring for `save_key_metric`: whether to save checkpoint or session when the value of ``key_metric`` is higher than all the previous values during training.

docstring for `key_metric_n_saved`: save top N checkpoints or sessions, sorted by the value of `key metric` in descending order.

Reply:
updated in the latest patch.
Thanks.